graspologic/cluster/autogmm.py

```diff
@@ -208,18 +210,18 @@ class AutoGMMCluster(BaseCluster):
 
     def __init__(
         self,
-        min_components=2,
-        max_components=10,
-        affinity="all",
-        linkage="all",
-        covariance_type="all",
-        random_state=None,
-        label_init=None,
-        max_iter=100,
-        verbose=0,
-        selection_criteria="bic",
-        max_agglom_size=2000,
-        n_jobs=None,
+        min_components: int = 2,
+        max_components: Optional[int] = 10,
+        affinity: Union[str, np.ndarray, List[str]] = "all",
+        linkage: Union[str, np.ndarray, List[str]] = "all",
+        covariance_type: Union[str, np.ndarray, List[str]] = "all",
+        random_state: Optional[Union[int, np.random.RandomState]] = None,
+        label_init: Optional[Union[np.ndarray, List[int]]] = None,
+        max_iter: int = 100,
+        verbose: int = 0,
```

Comment:
so these don't get `Optional` cause they can't be none?

Reply:
Precisely; describing a variable or parameter of a type of `x: Optional[int]` to most languages means `this value can exist and be an int or not exist entirely`.

Whereas in this context, we're saying the item will ALWAYS be an integer, and everything else is invalid.

That's why `Optional` is such a strong word to use in documentation; it is an encapsulating type that says "may exist or may not exist but if it exists it must be this type", whereas using it to describe a defaulted parameter tells the user they can pass in `None` to maybe opt out of _something_, but that would be an error.

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html
https://doc.rust-lang.org/std/option/
https://www.scala-lang.org/api/current/scala/Option.html
https://en.wikipedia.org/wiki/Option_type

Reply:
And more specifically: https://docs.python.org/3/library/typing.html#typing.Optional
> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. ... On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not.